### DISCUSSION TITLE
Функция craftTracks возвращает пустой массив

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#34] [Помощь с алгоритмом] [answered]
Хотел сравнить работу функции craftTracks по трекам и по исполнителям.

В описании функции есть следующий пример. Он должен работать?

Пример 1 - Получить рекомендации по всем любимым трекам по их исполнителям
```js
let tracks = Source.getSavedTracks();
let recomTracks = Source.craftTracks(tracks, {
    key: 'seed_artists',
    query: {
        min_energy: 0.4,
        min_popularity: 60,
        // target_popularity: 60,
    }
});
```

Потому что я вижу такой ответ и треков нет на выходе
```js
response { tracks: [],
  seeds: 
   [ { initialPoolSize: 0,
       afterFilteringSize: 0,
       afterRelinkingSize: 0,
       id: '2kCWxXQc7gB1hEbN0rcMHo',
       type: 'ARTIST',
       href: 'https://api.spotify.com/v1/artists/2kCWxXQc7gB1hEbN0rcMHo' },
     { initialPoolSize: 0,
       afterFilteringSize: 0,
       afterRelinkingSize: 0,
       id: '2xiIXseIJcq3nG7C8fHeBj',
       type: 'ARTIST',
       href: 'https://api.spotify.com/v1/artists/2xiIXseIJcq3nG7C8fHeBj' },
     { initialPoolSize: 0,
       afterFilteringSize: 0,
       afterRelinkingSize: 0,
       id: '4kCIzZ4lrPzHq8o18sDlSu',
       type: 'ARTIST',
       href: 'https://api.spotify.com/v1/artists/4kCIzZ4lrPzHq8o18sDlSu' } ] }
```

**Comment by Chimildic** [comment, marked as answer]
На моей стороне этот пример работает, треки возвращаются. Если касаться источника из любимых треков.
Возможно две причины.

Исходный массив треков состоит из малоизвестных или новых исполнителей/треков. В документации к API есть примечание, что для них недостаточно данных для создания рекомендаций. [Онлайн консоль](https://developer.spotify.com/console/get-recommendations/) тоже возвращает пустые рекомендации для перечисленных `id`.

Относительно недавно у Spotify были замечены проблемы с возвращаемыми рекомендациями. Даже по популярным исполнителям/трекам возвращался пустой список. Вот, например, [еще один](https://community.spotify.com/t5/Spotify-for-Developers/Technical-issues-with-Recommendations-API/m-p/5146203) свежий сигнал об этом. Мы ничего сделать не можем.

Больше склоняюсь ко второму варианту. Не смотря на то, что у перечисленных `id` мало подписчиков.

**Reply by efremych** [reply]
а, ясно, спасибо большое, получается, действительно, похоже на то что просто Спотифай не возвращает. 
При этом, "по трекам" - норм.
